### ISSUE TITLE
Some problems when running experiments

### ISSUE THREAD

**Issue by Neronjust2017** [danikiyasseh/CLOCS#1]
Hi, I'm very interested in your work. However, I encountered some problems when running your code. 
(1) when running CMSC(contrastive_ms, chapman),  I got this error:
![图片](https://user-images.githubusercontent.com/27863360/103146759-bdc71f80-4788-11eb-8881-430fdd26ee1c.png)
![图片](https://user-images.githubusercontent.com/27863360/103146782-fcf57080-4788-11eb-8973-2f48e4009a8d.png)
It seems that the correct shape of "input_frame" is (5000, ), but the actual shape is (4, 5000). But the data loaded here was generated by using the script "load_chapman_ecg.py", which you provided and the num of leads is 4. Here is the data folder:

├─contrastive_ml
│  └─leads_['II', 'V2', 'aVL', 'aVR']
├─contrastive_ms
│  └─leads_['II', 'V2', 'aVL', 'aVR']
├─contrastive_msml
│  └─leads_['II', 'V2', 'aVL', 'aVR']
├─contrastive_ss
│  └─leads_['II', 'V2', 'aVL', 'aVR']

And contrastive_ms\leads_['II', 'V2', 'aVL', 'aVR']\frames_phases_chapman.pkl :
![图片](https://user-images.githubusercontent.com/27863360/103146846-fddad200-4789-11eb-92df-669d1f5c12d6.png)
Is there something wrong in your code? or I missed something?

(2) When I run CMSMLC or CMLC, the problem above didn't occur anymore, cause CMSC only uses one lead and CMLC. CMSMLC use all leads. However, the losses(train and valid) are always NAN. I also used the pre-processing code you provided, no modifications. 
![图片](https://user-images.githubusercontent.com/27863360/103146904-c4ef2d00-478a-11eb-980e-9f015697c481.png)

Hope you can tell me how to solve this. Thanks very much!

**Comment by Rok07** [comment]
Hi,
I want to know how to pre-process the datasets for CLOCS.
Is there any idea for pre-processing with the given URL's code? 
Thank you

**Comment by danikiyasseh** [comment]
Hello, I address both of your questions / concerns below:

1) To clarify, CMSC uses however many leads you outline in the data-loading script. From a data-loading perspective, the difference between CMSC and CMLC, for example, is that the former collapses all leads into the first dimension of the data matrix. More concretely, the CMSC matrix should have dimensions (N x S) where N is the number of segments (6387 for training set) multiplied by the number of leads (4) and S is the length of the segment (e.g., 5000). The load-chapman-ecg.py script should do that for you (I will check again). In the meantime, you can simply flatten the first two dimensions that you have. 

2) The NaN loss, from my experience, is a consequence of a NaN value in the actual ECG segment. Although the load-chapman-ecg.py script had a NaN removal process, I have now updated this script with an offline version that should remove all ECG segments with NaN values. 

I hope these make sense - please tell me if you are still facing issues.
